src/cstring.rs

```diff
@@ -0,0 +1,41 @@
+use std::{ffi::{self}, fmt};
+
+use serde::{de::{self, Visitor}, Deserializer, Serializer};
+
+
+pub type CString = ffi::CString;
+
+pub fn serialize<S>(string: &CString, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&string.to_string_lossy())
+}
+
+
+struct CStringVisitor;
+
+impl Visitor<'_> for CStringVisitor {
+    type Value = CString;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a string")
```

Comment:
try the write macro, it works like the print(ln) macro:

```suggestion
        write!(formatter, "a string")
```